src/docgen/schema.ts

```diff
@@ -0,0 +1,465 @@
+   */
+  readonly constants: PropertySchema[];
+}
+
+/**
+ * Describes a construct.
+ */
+export interface ConstructSchema extends ClassSchema {}
+
+/**
+ * Describes a struct.
+ */
+export interface StructSchema extends Usage, Documentable {
+  /**
+   * An id that uniquely identifies this struct among all entities in
+   * the document and is same across languages.
+   */
+  readonly id: string;
+
+  /**
+   * The friendly language-specific name for the struct.
+   */
+  readonly displayName: string;
+
+  /**
+   * The language-specific fqn.
+   */
+  readonly fqn: string;
+
+  /**
+   * Properties.
+   */
+  readonly properties: PropertySchema[];
+}
+
+/**
+ * Describes a behavioral interface, also sometimes known as a protocol.
+ */
+export interface InterfaceSchema extends Documentable {
+  /**
+   * An id that uniquely identifies this interface among all entities in
+   * the document and is same across languages.
+   */
+  readonly id: string;
+
+  /**
+   * The friendly language-specific name for the interface.
+   */
+  readonly displayName: string;
+
+  /**
+   * The language-specific fqn.
+   */
+  readonly fqn: string;
+
+  /**
+   * Interfaces that this interface extends.
+   */
+  readonly interfaces: JsiiEntity[];
+
+  /**
+   * Types implementing this interface.
+   */
+  readonly implementations: JsiiEntity[];
+
+  /**
+   * Methods.
+   */
+  readonly instanceMethods: MethodSchema[];
+
+  /**
+   * Properties.
+   */
+  readonly properties: PropertySchema[];
+}
+
+/**
+ * Describes an enum member.
+ */
+export interface EnumMemberSchema extends Documentable {
+  /**
+   * An id that uniquely identifies this enum member among all entities in
+   * the document and is same across languages.
+   */
+  readonly id: string;
+
+  /**
+   * The friendly language-specific name for the enum member.
+   */
+  readonly displayName: string;
+
+  /**
+   * The language-specific fqn.
+   */
+  readonly fqn: string;
+}
+
+/**
+ * Describes an enum.
+ */
+export interface EnumSchema extends Documentable {
+  /**
+   * An id that uniquely identifies this enum among all entities in
+   * the document and is same across languages.
+   */
+  readonly id: string;
+
+  /**
+   * The friendly language-specific name for the enum.
+   */
+  readonly displayName: string;
+
+  /**
+   * The language-specific fqn.
+   */
+  readonly fqn: string;
+
+  /**
+   * Enum members.
+   */
+  readonly members: EnumMemberSchema[];
+}
+
+/**
+ * Describes the API Reference.
+ */
+export interface ApiReferenceSchema {
+
+  /**
+   * Constructs.
+   */
+  readonly constructs: ConstructSchema[];
+
+  /**
+   * Classes.
+   */
+  readonly classes: ClassSchema[];
+
+  /**
+   * Structs.
+   */
+  readonly structs: StructSchema[];
+
+  /**
+   * Interfaces.
+   */
+  readonly interfaces: InterfaceSchema[];
+
+  /**
+   * Enums.
+   */
+  readonly enums: EnumSchema[];
+}
+
+/**
+ * Metadata about a particular jsii assembly.
+ */
+export interface AssemblyMetadataSchema {
```

Comment:
I feel like we should add a `language` property here. Consumers should be able to tell which language they are looking at.